db/schema_tables.cc

```diff
@@ -773,19 +773,18 @@ schema_ptr scylla_table_schema_history() {
 #endif
 
 static
```

Comment:
>schema_tables: redact_columns_for_missing_features: upgrade mutation …
…in place

> When needed, the input mutation can be upgraded
> in place instead being copied and modified,
> to be returned as the function's result value.

1. missing motivation
2. seems a regression in readabilty - an in/out variable is less clear at the call site than a return value

Reply:
The motivation is reducing moves, though they are cheap as they only move a unique_ptr holding the mutation data.
This was particularly weird, in my eyes in:
```
adjust_schema_for_schema_features(std::vector<mutation> schema, schema_features features) {
    for (auto& m : schema) {
        m = redact_columns_for_missing_features(std::move(m), features);
    }
    return schema;
}
```

Which clearly modifies the mutations in place (as are all other call sites, but there it's less obvious).

Reply:
Why weird? It's clear what the input and output are.